pymatgen/entries/compatibility.py

```diff
@@ -407,9 +411,127 @@ def __str__(self):
         return "{} {} Correction".format(self.name, self.compat_type)
 
 
-class Compatibility(MSONable):
+class Compatibility(MSONable, metaclass=abc.ABCMeta):
     """
-    The Compatibility class combines a list of corrections to be applied to
+    Abstract Compatibility class, not intended for direct use.
+    Compatibility classes are used to correct the energies of an entry or a set
+    of entries. All Compatibility classes must implement .get_adjustments method.
+    """
+    @abc.abstractmethod
+    def get_adjustments(self, entry: ComputedEntry):
+        """
+        Get the energy adjustments for a ComputedEntry.
+
+        This method must generate a list of EnergyAdjustment objects
+        of the appropriate type (constant, composition-based, or temperature-based)
+        to be applied to the ComputedEntry, and must raise a CompatibilityError
+        if the entry is not compatible.
+
+        Args:
+            entry: A ComputedEntry object.
+
+        Returns:
+            [EnergyAdjustment]: A list of EnergyAdjustment to be applied to the
+                Entry.
+
+        Raises:
+            CompatibilityError if the entry is not compatible
+        """
+        return
+
+    def process_entries(self, entries: Union[ComputedEntry, list], clean: bool = False):
+        """
+        Process a sequence of entries with the chosen Compatibility scheme.
+
+        Args:
+            entries: ComputedEntry or [ComputedEntry]
+            clean: bool, whether to remove any previously-applied energy adjustments.
+                If True, all EnergyAdjustment are removed prior to processing the Entry.
+                Default is False.
+
+        Returns:
+            A list of adjusted entries.  Entries in the original list which
+            are not compatible are excluded.
+        """
+        # convert input arg to a list if not already
+        if isinstance(entries, ComputedEntry):
+            entries = [entries]
+
+        processed_entry_list = []
+
+        if clean:
+            for e in entries:
+                # remove any previously documented corrections from the entry
+                e.energy_adjustments = []
+
+        # apply the corrections
+        for entry in entries:
+            try:
+                adjustments = self.get_adjustments(entry)
+                for ea in adjustments:
+                    # Has this correction already been applied?
+                    if ea.name in [e.name for e in entry.energy_adjustments]:
+                        if ea.value in [e.value for e in entry.energy_adjustments]:
```

Comment:
What if you have two of the same correction with different values, but one of the values also is from another correction? That would fail this check.

Reply:
See revised code based on comparing tuples of (name, cls, value) and (name, cls).